compose/config/config.py

```diff
@@ -345,14 +345,14 @@ def validate_extended_service_dict(service_dict, filename, service):
                 "%s services with 'net: container' cannot be extended" % error_prefix)
 
 
-def process_container_options(service_dict, working_dir=None):
-    service_dict = service_dict.copy()
+def process_container_options(working_dir, service_dict):
+    service_dict = dict(service_dict)
```

Comment:
`dict()` would also make something like `[(1, 2)]` into `{ 1: 2 } `, transforming potentially garbage data into a dictionary, while copy() would fail there. It's probably irrelevant, so I'm fine with the change.